controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -534,6 +534,20 @@ func (r *BareMetalHostReconciler) actionDeleting(prov provisioner.Provisioner, i
 		return actionError{errors.Wrap(err, "failed to remove finalizer")}
 	}
 
+	if utils.StringInList(info.bmcCredsSecret.Finalizers, metal3v1alpha1.BareMetalHostFinalizer+"/secret") {
+		// Remove finalizer from secret to allow deletion
+		info.bmcCredsSecret.Finalizers = utils.FilterStringFromList(
+			info.bmcCredsSecret.Finalizers, metal3v1alpha1.BareMetalHostFinalizer+"/secret")
+
+		if err := r.Update(context.Background(), info.bmcCredsSecret); err != nil {
+			return actionError{errors.Wrap(err, fmt.Sprintf("failed to remove finalizer from secret %s in namespace %s",
+				info.bmcCredsSecret.ObjectMeta.Name, info.bmcCredsSecret.ObjectMeta.Namespace))}
+		}
+
+		info.log.Info("removed secret finalizer",
+			"remaining", info.bmcCredsSecret.Finalizers)
+	}
```

Comment:
Please move this implementation to the SecretManager as a `ReleaseSecret()` method or something like that.